### DISCUSSION TITLE
module 'umap' has no attribute 'UMAP'

### DISCUSSION THREAD

**Discussion by aelhossiny** [lmcinnes/umap#589] [Q&A]
I keep getting this error when I try to plot my data. 

`dir(umap)` gives the following

`['__builtins__',
 '__cached__',
 '__doc__',
 '__file__',
 '__loader__',
 '__name__',
 '__package__',
 '__path__',
 '__spec__',
 'distances',
 'layouts',
 'sparse',
 'spectral',
 'utils']`

**Comment by lmcinnes** [comment]
That's very odd! Somehow it has lost the modules and imports it is supposed to have. I am honestly at a loss as to how or why this could occur. Is there anyone with more expertise in python module handling that can help out here?

**Reply by aelhossiny** [reply]
I guess it's something with the university cluster. It works fine on my machine!

**Reply by aelhossiny** [reply]
It's resolved now when I installed the latest version of numba

`pip install --user numba==0.52`

Thanks!
